packages/monaco/src/browser/monaco-editor-provider.ts

```diff
@@ -341,10 +346,11 @@ export class MonacoEditorProvider {
         toDispose.push(editor.onLanguageChanged(() => this.updateMonacoDiffEditorOptions(editor)));
         return editor;
     }
-    protected createMonacoDiffEditorOptions(original: MonacoEditorModel, modified: MonacoEditorModel): MonacoDiffEditor.IOptions {
+    protected createMonacoDiffEditorOptions(original: MonacoEditorModel, modified: MonacoEditorModel, factoryOptions?: EditorFactoryOptions): MonacoDiffEditor.IOptions {
         const options = this.createOptions(this.diffPreferencePrefixes, modified.uri, modified.languageId);
         options.originalEditable = !original.readOnly;
         options.readOnly = modified.readOnly;
+        options.alwaysRevealFirst = Boolean(factoryOptions?.selection);
```

Comment:
This is really the only operative line in the whole PR.

I don't see any particular reason why the `monaco` package shouldn't know about the options used in `editor` to create `EditorWidget`s, but it is possible to do all the work in `monaco`, but it requires manipulating some `readonly` / `private` fields on the underlying DiffNavigator in response to `set cursor` or `set selection` calls.